proto/encode.go

```diff
@@ -189,8 +189,22 @@ type Marshaler interface {
 // prefixed by a varint-encoded length.
 func (p *Buffer) EncodeMessage(pb Message) error {
 	siz := Size(pb)
+	sizVar := SizeVarint(uint64(siz))
```

Comment:
Shouldn't we be fixing `p.Marshal` rather than the calling code? If we have to change code that calls `Marshal`, then I suspect something must be wrong?

Reply:
I think you are right. The large change here isn't necessary